jax/_src/scipy/sparse/linalg.py

```diff
@@ -277,15 +272,37 @@ def cg(A, b, x0=None, *, tol=1e-5, atol=0.0, maxiter=None, M=None):
       inverse of A.  Effective preconditioning dramatically improves the
       rate of convergence, which implies that fewer iterations are needed
       to reach a given error tolerance.
+  assume_ipart_is_zero : bool, optional
+      Whether the linear operator dtype can be assumed to have imag(A)=0.
+      Defaults to False for complex-value systems. When True, then it can be
+      assumed that for complex operators A, A @ x and A.T @ x are equivalent,
+      leading to more efficient reverse-mode autodiff.
 
   See also
   --------
   scipy.sparse.linalg.cg
   jax.lax.custom_linear_solve
   """
+  if assume_ipart_is_zero:
+    # Don't need to check for symmetry if we assume the imaginary part is zero.
+    symmetric = True
+  else:
+    # real-valued positive-definite linear operators are symmetric.
+    def real_valued(x):
+      return not issubclass(jnp.result_type(x).type, np.complexfloating)
```

Comment:
This is the more idiomatic way to write this check:
```suggestion
      return not dtypes.issubdtype(jnp.result_type(x), np.complexfloating)
```